azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -197,3 +197,20 @@ $actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$d
 if (-not ($actual.Contains("Asset cache hit for example3.html; downloaded from: file://$AssetCache"))) {
     throw "Failure: azurl (yes), x-block-origin (yes), asset-cache (hit), download (n/a)"
 }
+
+# Testing x-download failure with asset cache (x-script) and x-block-origin settings
+$env:X_VCPKG_ASSET_SOURCES = "clear;x-script,pwsh $PSScriptRoot/../e2e-assets/asset-caching/failing-script.ps1 {url} {sha512} {dst};x-block-origin"
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--url", "https://example.com", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a"))
+if (-not ($actual.Contains("error: <mirror-script> failed with exit code: (1).") -and
+          $actual.Contains("error: Missing example3.html and downloads are blocked by x-block-origin."))) {
```

Comment:
Since you are trying to check the order of these, should the test actually check the order?